Classes/Middleware/RedirectHandler.php

```diff
@@ -117,6 +117,11 @@ protected function buildRedirectResponse(UriInterface $uri, array $redirectRecor
             $frontendDomainTrim = false;
             $targetUrl = $resolvedTarget['file'];
         } else {
+            if (substr($uri->getPath(), 0, 3) === '%7B') {
```

Comment:
I'm not sure if it's the best way to create condition against. Maybe there is another way to know if it's record links?

Reply:
@lukaszuznanski, surely not ;-) i just wanted to open a discussion about it, see https://github.com/TYPO3-Initiatives/headless/issues/314#issuecomment-920668994